src/main/java/com/avispl/symphony/dal/infrastructure/management/neat/pulse/NeatPulseCommunicator.java

```diff
@@ -794,7 +878,7 @@ private void retrieveSystemInfo() throws Exception {
 			if (response != null && response.has(NeatPulseConstant.ENDPOINTS) && response.get(NeatPulseConstant.ENDPOINTS).isArray()) {
 				deviceList.clear();
 				for (JsonNode node : response.get(NeatPulseConstant.ENDPOINTS)) {
-					deviceList.add(node.get(NeatPulseConstant.ID).asText());
+					deviceList.put(node.get(NeatPulseConstant.ID).asText(), node.get("roomId").asText());
```

Comment:
same here and a few other .get() calls down the line to avoid eventual NPEs
It may be ok if we can guarantee the node will not result in null values though, so use common sense :)